app/controllers/api/v1/room_settings_controller.rb

```diff
@@ -15,12 +15,20 @@ def show
       # PATCH /api/v1/room_settings/:friendly_id
       def update
         config = MeetingOption.get_config_value(name: room_setting_params[:settingName], provider: current_provider)&.value
+        return render_error status: :bad_request unless config
 
-        return render_error status: :forbidden unless config == 'optional'
+        name = room_setting_params[:settingName]
+        value = room_setting_params[:settingValue].to_s
 
-        option = @room.get_setting(name: room_setting_params[:settingName])
+        is_access_code = %w[glViewerAccessCode glModeratorAccessCode].include? name
 
-        return render_error status: :bad_request unless option&.update(value: room_setting_params[:settingValue].to_s)
+        return render_error status: :forbidden if config == 'false' || (config == 'true' && (!is_access_code || value == 'false'))
```

Comment:
This may be hard to fallow, it's a result of a mathematical simplification (Boolean Algebra) of the fallowing expression:
```
if (is_access_code && value != 'false' && config == 'false') || (is_access_code && value == 'false' && config != 'optional') || (!is_access_code && config != 'optional')
  return render_error status: :forbidden # Re-generting access code is only forbidden when it's disabled.
end
```

First simplification:
```
 if (is_access_code && value != 'false' && config == 'false') || (config != 'optional' && (!is_access_code || value == 'false'))
   return render_error status: :forbidden # Re-generting access code is only forbidden when it's disabled.
end
```
...

Simplified version:
```
if config == 'false' || (config == 'true' && (!is_access_code || value == 'false'))
  return render_error status: :forbidden # Re-generting access code is only forbidden when it's disabled.
end
```

Reply:
Couldn't you just do `if config != 'optional' || !(is_access_code && config == 'true' && value != 'false')

Reply:
I believe `if config != 'optional' || !(is_access_code && config == 'true' && value != 'false')` is the same as:
`if config != 'optional' || !is_access_code || config != 'true' || value == 'false'`.
Unfortunately, this is not the case.
This fails for cases like the option isn't an access code and it's optional, an optional forced enabled access code,...
Plus it's not order independent,

Reply:
I may be wrong, will further discuss it in our meeting today

Reply:
How about this? This way we make it clear the 2 conditions to get through are either optional config or force enabled access code

`if !(config == optional || (config == true && is_access_code && value == true))`

Reply:
Yup, that's 🔥 
It's a lot more readable,
I'm just make this minor changes because we're mapping only `value=='false'` explicitly,
`unless config == 'optional' || (config == 'true' && is_access_code && value != 'false')`